2020-02-Spring/3_trees/README.md

```diff
@@ -141,21 +211,287 @@ Go to [Top](#top)
 <!-- Don't remove -->
 <a name="s3"/>
 
-### 3. SOLUTION 3 TODO :bug:
+### 3. Recover Binary Search Tree Solution
 
-Source: TODO :bug:
+#### Basic Solution 
 
-#### Naive/Simple Solution 
+<details>
+<summary>Click to see basic solution</summary>
+
+##### Algorithm Overview
+
+The key idea behind this problem is using the properties of the inorder traversal of a binary search tree to identify the nodes that you must fix.
+
+1. Create an inorder traversal of the tree. This will be an almost sorted list.
+2. Find two nodes in the almost sorted list that are not in sorted order.
+3. Traverse the tree and fix the nodes that are not in sorted order.
```

Comment:
Ah so the point is that it should be in sorted order. Disregard above. I would not have guessed that right away. Is this something you'll tell them if they ask during the session?